plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -71,6 +85,92 @@ public function getKnownSegmentsToArchiveForSite(&$segments, $idSite)
         $segments = array_unique($segments);
     }
 
+    public function onNoArchiveData()
+    {
+        $segmentInfo = $this->getSegmentIfIsUnprocessed();
+        if (empty($segmentInfo)) {
+            return;
+        }
+
+        list($segment, $isSegmentToPreprocess) = $segmentInfo;
+
+        // this archive has no data, the report is for a segment that gets preprocessed, and the archive for this
+        // data does not exist. this means the data will be processed later. we let the user know so they will not
+        // be confused.
+        $model = new Model();
+        $storedSegment = $model->getSegmentByDefinition($segment->getString()) ?: null;
+
+        throw new UnprocessedSegmentException($segment, $isSegmentToPreprocess, $storedSegment);
+    }
+
+    public function onLoadingError(\Exception $ex, View $dataTableView)
+    {
+        if (!($ex instanceof UnprocessedSegmentException)) {
+            Notification\Manager::cancel(self::NO_DATA_UNPROCESSED_SEGMENT_ID);
```

Comment:
What if there are multiple reports on the page and eg the first fails because of unprocessed and the next one because of some other reason? Then the notification wouldn't be sent?

Reply:
That was just to be safe, but I guess it doesn't really help, I'll remove it.